analytics/src/main/java/com/segment/analytics/internal/AnalyticsClient.java

```diff
@@ -227,22 +269,28 @@ public void run() {
               log.print(VERBOSE, "Flushing messages.");
             }
           } else {
+            messageIdSizeMap.put(message.messageId(), messageSizeInBytes(message));
             messages.add(message);
           }
 
           Boolean isBlockingSignal = message == FlushMessage.POISON || message == StopMessage.STOP;
           Boolean isOverflow = messages.size() >= size;
 
           if (!messages.isEmpty() && (isOverflow || isBlockingSignal)) {
-            Batch batch = Batch.create(CONTEXT, messages);
-            log.print(
-                VERBOSE,
-                "Batching %s message(s) into batch %s.",
-                messages.size(),
-                batch.sequence());
-            networkExecutor.submit(
-                BatchUploadTask.create(AnalyticsClient.this, batch, maximumRetries));
-            messages = new ArrayList<>();
+            while (!messages.isEmpty()) {
+              Batch batch =
+                  BatchUtility.createBatch(
+                      messages, messageIdSizeMap, contextSize, sequenceCounter);
```

Comment:
no need to keep a `sequenceCounter`, we can just use the maximum length of string of the max integer value (`String.valueOf(Integer.MAX_INT).length`) as the length of the counter.

Reply:
ohh you mean just consider the longest possible int value so we dont keep a count of the sequenceCounter right? 

yeah that could work too. Thanks!